js/viz/gauges/utils/resolve_label_overlapping.js

```diff
@@ -0,0 +1,104 @@
+        const angle = normalizeAngle(bar._angle);
+        bar._text._lastCoords = { x: 0, y: 0 };
+        (angle <= 90 || angle >= 270 ? r.right : r.left)
+            .push({
+                series: {
+                    isStackedSeries: ()=> false,
+                    isFullStackedSeries: ()=> false
+                },
+                getLabels: ()=> [{
+                    isVisible: ()=> true,
+                    getBoundingRect: () => {
+                        const { height, width, x, y } = bar._text.getBBox();
+                        const lastCoords = bar._text._lastCoords;
+
+                        return {
+                            x: x + lastCoords.x,
+                            y: y + lastCoords.y,
+                            width: width,
+                            height: height,
+                        };
+                    },
+                    shift: (x, y)=> {
+                        const box = bar._text.getBBox();
+
+                        bar._text._lastCoords = { x: x - box.x, y: y - box.y };
+                        bar._text.attr({ translateX: x - box.x, translateY: y - box.y });
+                    },
+                    draw: ()=> bar.hideLabel(),
+                    getData: ()=> { return { value: bar.getValue() }; },
+                    hideInsideLabel: ()=> false,
+                }]
+            }
+            );
+        return r;
+    }, { left: [], right: [] });
+}
+
+export const clearLabelsCrossTitle = (bars, minY)=> {
+    bars.forEach(bar => {
+        const box = bar._text.getBBox();
+        const lastCoords = bar._text._lastCoords;
+        if(minY > box.y + lastCoords.y) {
+
+            bar.hideLabel();
+        }
+    });
+};
+
+const getStartCoordsArc = function(x, y, innerR, outerR, startAngleCos, startAngleSin, endAngleCos, endAngleSin, isCircle, longFlag) {
+    return { x: (x + outerR * startAngleCos).toFixed(ARC_COORD_PREC), y: (y - outerR * startAngleSin).toFixed(ARC_COORD_PREC) };
+};
+
+export const drawConnector = (bars, connectorWidth) => {
+    bars.forEach((bar) => {
+        const x = bar._bar.attr('x');
+        const y = bar._bar.attr('y');
+        const innerRadius = bar._bar.attr('innerRadius');
+        const outerRadius = bar._bar.attr('outerRadius');
+        const startAngle = bar._bar.attr('startAngle');
+        const endAngle = bar._bar.attr('endAngle');
+        const coordStart = getStartCoordsArc.apply(null, normalizeArcParams(x, y, innerRadius, outerRadius, startAngle, endAngle));
+        const xStart = Number(coordStart.x);
+        const yStart = Number(coordStart.y);
+        const box = bar._text.getBBox();
+        const shiftConnector = connectorWidth / 2;
+        const lastCoords = bar._text._lastCoords;
+
+        if(bar._angle > 90) {
+            bar._line.attr({ points: [
+                xStart + shiftConnector, yStart + shiftConnector, box.x + box.width + lastCoords.x, box.y + box.height / 2 + lastCoords.y
```

Comment:
1. It's not necessary to do it on the single row.
2. Seems like this points and points on the `else` part differs only by the first point. Can we add all other after this confition?

Reply:
No, they differ because of the calculation 
I made line breaks, now it is clear that only the "y" coordinate of the second point is the same